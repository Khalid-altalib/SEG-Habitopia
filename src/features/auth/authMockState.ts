const authMockState = {
  loading: false,
  error: ""
};

export default authMockState;