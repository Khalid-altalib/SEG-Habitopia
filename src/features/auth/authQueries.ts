import { GraphQLQuery } from "@aws-amplify/api";
import { API, Auth, DataStore } from "aws-amplify";
import { User } from "../../models";
import { CreateUserInput, CreateUserMutation } from "../../API";
import { createUser } from "../../graphql/mutations";

export const createUserInDatabase = async (
  userId: string,
  email?: string,
  name?: string
) => {
  const userQuery = await DataStore.query(User, (user) => user.id.eq(userId));
  if (userQuery.length == 0) {
    try {
      const user: CreateUserInput = { id: userId, name: name, email: email };
      await API.graphql<GraphQLQuery<CreateUserMutation>>({
        query: createUser,
        variables: { input: user },
      });
    } catch (error) {
      console.log(error);
    }
  }
};
