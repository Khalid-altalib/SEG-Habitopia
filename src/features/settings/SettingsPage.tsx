import React from "react";
import FlowLayout from "../../components/FlowLayout/FlowLayout";
import { Input, Text, VStack } from "native-base";

const SettingsPage = () => {
  return (
    <FlowLayout>
      <Text mt={5}>Account Information</Text>
      <VStack mt={8}>
        <Text fontWeight={"bold"}>Phone Number</Text>
        <Input />
      </VStack>
      <VStack mt={8}>
        <Text fontWeight={"bold"}>Email Address</Text>
        <Input />
      </VStack>
      <VStack mt={8}>
        <Text fontWeight={"bold"}>Password</Text>
        <Input secureTextEntry={true} />
      </VStack>
      <Text style={{ fontWeight: "bold", marginTop: 16 }}>
        Notification Preferences
      </Text>
      <VStack mt={8}>
        <Text fontWeight={"bold"}>Notification Settings</Text>
        <Input />
      </VStack>
      <Text style={{ fontWeight: "bold", marginTop: 16 }}>
        Billing Information
      </Text>
      <VStack mt={8}>
        <Text fontWeight={"bold"}>Billing Information</Text>
        <Input />
      </VStack>
    </FlowLayout>
  );
};

export default SettingsPage;
