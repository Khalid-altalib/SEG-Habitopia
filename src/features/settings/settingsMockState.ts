import { SettingsState } from "./settingsSlice";

/**
 * The mock state for the settings redux state
 */

const settingsMockState: SettingsState = {
  settings: {
    email: "test@test.com",
    password: "password",
    notifications: true,
    name: "Test User",
    biography: "Hello, I'm Test User.",
  },
  fetchSettings: {
    loading: false,
    error: "",
  },
  setSettings: {
    loading: false,
    error: "",
  },
};

export default settingsMockState;
