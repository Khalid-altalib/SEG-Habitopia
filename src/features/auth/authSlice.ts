import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
import type { PayloadAction } from "@reduxjs/toolkit";
import { LocalUser } from "../../../types";
import AsyncStorage from "@react-native-async-storage/async-storage";
import { RootState } from "../../app/store";
import { Toast } from "react-native-toast-message/lib/src/Toast";
import { createUserInDatabase } from "./authQueries";
import { Auth } from "aws-amplify";

type AuthState = {
  signUpData: {
    email: string;
    password: string;
    name: string;
    confirmationCode: string;
  };
  logInData: {
    email: string;
    password: string;
  };
  user: LocalUser | undefined;
  loading: boolean;
  error: string;
};

export const signUpUser = createAsyncThunk<void, void, { rejectValue: string }>(
  "auth/signUp",
  async (_, thunkAPI) => {
    try {
      const state = thunkAPI.getState() as RootState;
      const { email, password, name } = state.auth.signUpData;
      await Auth.signUp({
        username: email as string,
        password: password as string,
        attributes: {
          name: name,
        },
      });
    } catch (error: any) {
      const message = error.message;
      console.log(error);
      return thunkAPI.rejectWithValue(message);
    }
  }
);

export const sendConfirmationCode = createAsyncThunk<
  LocalUser,
  void,
  { rejectValue: string }
>("auth/confirmation", async (_, thunkAPI) => {
  try {
    const state = thunkAPI.getState() as RootState;
    const { email, password, confirmationCode, name } = state.auth.signUpData;
    await Auth.confirmSignUp(email, confirmationCode);
    const user = await logInHelper(email, password, name);
    return user as LocalUser;
  } catch (error: any) {
    const message = error.message;
    console.log(error);
    return thunkAPI.rejectWithValue(message);
  }
});

export const updatePassword = async (password: string, thunkAPI: any) => {
  console.log("updating password");
  console.log(password);
  const oldPassword = "Password.123";
  const state = thunkAPI.getState() as RootState;
  // console.log(state);
  const email = state.settings.settings.email;
  console.log(email);

  Auth.currentAuthenticatedUser()
  .then((user) => {
    return Auth.changePassword(user, oldPassword, password);
  })
  .then((data) => console.log(data))
  .catch((err) => console.log(err));
  
};

const logInHelper = async (email: string, password: string, name?: string) => {
  const { signInUserSession, attributes } = await Auth.signIn(email, password);
  const user = {
    authToken: signInUserSession.idToken.jwtToken,
    userId: attributes.sub,
  };

  await AsyncStorage.setItem("user", JSON.stringify(user));
  await createUserInDatabase(attributes.sub, email, name);
  return user;
};

export const logInUser = createAsyncThunk<
  LocalUser,
  void,
  { rejectValue: string }
>("auth/login", async (_, thunkAPI) => {
  try {
    const state = thunkAPI.getState() as RootState;
    const { email, password } = state.auth.logInData;
    const user = await logInHelper(email, password);
    return user as LocalUser;
  } catch (error: any) {
    const message = error.message;
    console.log(message);
    return thunkAPI.rejectWithValue(message);
  }
});

export const logInUserFromStorage = createAsyncThunk<
  LocalUser,
  void,
  { rejectValue: string }
>("auth/logInFromStorage", async (_, thunkAPI) => {
  const data = await AsyncStorage.getItem("user");

  if (data) {
    Toast.show({
      type: "success",
      text1: "Welcome back! ðŸ‘‹",
      text2: "We've logged you in from your previous session",
    });
    return JSON.parse(data) as LocalUser;
  } else {
    return thunkAPI.rejectWithValue("Please log in again");
  }
});

export const logOutUser = createAsyncThunk("auth/logOutUser", async () => {
  await AsyncStorage.removeItem("user");
  return undefined;
});

const initialState: AuthState = {
  signUpData: { email: "", password: "", name: "", confirmationCode: "" },
  logInData: { email: "", password: "" },
  user: undefined,
  loading: false,
  error: "",
};

const authenticationPending = (state: AuthState) => {
  state.loading = true;
};

const authenticationFulfilled = (
  state: AuthState,
  action: PayloadAction<LocalUser>
) => {
  state.loading = false;
  state.user = action.payload;
  state.error = "";
};

const authenticationRejected = (state: AuthState, action: any) => {
  state.loading = false;
  state.user = undefined;
  state.error = action.payload;
};

export const authSlice = createSlice({
  name: "auth",
  initialState,
  reducers: {
    addSignUpData: (state, action: PayloadAction<object>) => {
      console.log(state, action);
      state.signUpData = { ...state.signUpData, ...action.payload };
    },
    addLogInData: (state, action: PayloadAction<object>) => {
      state.logInData = { ...state.logInData, ...action.payload };
    },
  },
  extraReducers: (builder) => {
    builder.addCase(logInUser.pending, authenticationPending);
    builder.addCase(logInUser.fulfilled, authenticationFulfilled);
    builder.addCase(logInUser.rejected, authenticationRejected);
    builder.addCase(sendConfirmationCode.pending, authenticationPending);
    builder.addCase(sendConfirmationCode.fulfilled, authenticationFulfilled);
    builder.addCase(sendConfirmationCode.rejected, authenticationRejected);
    builder.addCase(logInUserFromStorage.pending, authenticationPending);
    builder.addCase(logInUserFromStorage.fulfilled, authenticationFulfilled);
    builder.addCase(logInUserFromStorage.rejected, authenticationRejected);
    builder.addCase(logOutUser.pending, (state) => {
      state.loading = true;
    });
    builder.addCase(logOutUser.fulfilled, (state) => {
      state.loading = false;
      state.user = undefined;
    });
  },
});

export const { addSignUpData, addLogInData } = authSlice.actions;

export default authSlice.reducer;
