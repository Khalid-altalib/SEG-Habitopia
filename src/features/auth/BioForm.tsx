import React from "react";
import { Input, TextArea } from "native-base";
import { useForm, Controller } from "react-hook-form";
import { AntDesign } from "@expo/vector-icons";

import { addSignUpData } from "./authSlice";
import { useDispatch } from "../../app/hooks";
import Button from "../../components/Button";
import Text from "../../components/Text";
import { AuthParams, ButtonType, TextType } from "../../../types";
import { View } from "react-native";
import { NavigationProp, useNavigation } from "@react-navigation/native";

type formData = {
  bio: string;
};

const BioForm = () => {
  const { control, handleSubmit } = useForm({
    defaultValues: {
      bio: "",
    },
  });
  const dispatch = useDispatch();

  const onSubmit = async (data: formData) => {
    dispatch(addSignUpData(data));
    navigation.navigate("SignIn");
  };

  const navigation = useNavigation<NavigationProp<AuthParams>>();

  return (
    <View>
      <Text style={{ marginBottom: 5 }} type={TextType.Subtle}>
        Biography
      </Text>
      <Controller
        control={control}
        render={({ field: { onChange, value } }) => (
          <TextArea
            autoCompleteType={false}
            onChangeText={onChange}
            value={value}
            size="xl"
            maxLength={100}
            maxHeight={35 * 3}
            overflow="true"
          />
        )}
        name="bio"
      />
      <Button
        onPress={handleSubmit(onSubmit)}
        type={ButtonType.Primary}
        style={{ marginTop: 20 }}
        isFullWidth
        icon={<AntDesign name="arrowright" size={20} color="white" />}
      >
        Continue
      </Button>
    </View>
  );
};

export default BioForm;
