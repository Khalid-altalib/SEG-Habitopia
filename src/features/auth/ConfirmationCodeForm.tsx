import React from "react";
import { Input } from "native-base";
import { useForm, Controller } from "react-hook-form";
import { AntDesign } from "@expo/vector-icons";
import { NavigationProp, useNavigation } from "@react-navigation/native";

import { View } from "react-native";

import { addSignUpData, sendConfirmationCode } from "./authSlice";
import { useAppDispatch } from "../../app/hooks";
import Button from "../../components/Button";
import Text from "../../components/Text";
import { AuthParams, ButtonType, TextType } from "../../../types";

type formData = {
  confirmationCode: string;
};

const ConfirmationCodeForm = () => {
  // Initial form state
  const { control, handleSubmit } = useForm({
    defaultValues: {
      confirmationCode: "",
    },
  });

  // onSubmit handler
  const dispatch = useAppDispatch();

  const onSubmit = async (data: formData) => {
    dispatch(addSignUpData(data));
    await dispatch(sendConfirmationCode());
    navigation.navigate("SignIn");
  };

  // React Navigation
  const navigation = useNavigation<NavigationProp<AuthParams>>();

  // JSX
  return (
    <View>
      <Text style={{ marginBottom: 5 }} type={TextType.Subtle}>
        Confirmation Code
      </Text>
      <Controller
        control={control}
        render={({ field: { onChange, value } }) => (
          <Input onChangeText={onChange} value={value} size="xl" />
        )}
        name="confirmationCode"
      />
      <Button
        onPress={handleSubmit(onSubmit)}
        type={ButtonType.Primary}
        style={{ marginTop: 20 }}
        isFullWidth
        icon={<AntDesign name="arrowright" size={20} color="white" />}
      >
        Continue
      </Button>
    </View>
  );
};

export default ConfirmationCodeForm;
