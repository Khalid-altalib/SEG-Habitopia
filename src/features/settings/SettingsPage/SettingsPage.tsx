import React, { useEffect } from "react";
import { Box, VStack, Button } from "native-base";
import SettingsItem from "../SettingsItem/SettingsItem";
import NotificationToggle from "../NotificationToggle/NotificationToggle";
import { useAppDispatch, useAppSelector } from "../../../app/hooks";
import { logOutUser } from "../../auth/authSlice";
import StatusContainer from "../../../components/StatusContainer/StatusContainer";
import { fetchSettings } from "../settingsSlice";
import settingEntries from "../settingEntries";

const SettingsPage = () => {
  const { settings, fetchSettings: requestStatus } = useAppSelector(
    (state) => state.settings
  );

  const dispatch = useAppDispatch();

  useEffect(() => {
    dispatch(fetchSettings());
  }, []);

  return (
    <StatusContainer
      loading={requestStatus.loading}
      error={requestStatus.error}
      data={settings.email.length > 0}
    >
      <VStack>
        {Object.keys(settingEntries).map((key) => (
          <Box paddingBottom="3" key={key}>
            <SettingsItem
              type={key}
              entry={settingEntries[key]}
              value={settings[key] as string}
            />
          </Box>
        ))}
        <NotificationToggle defaultValue={settings.notifications} />
        <Button
          onPress={() => {
            dispatch(logOutUser());
          }}
          my={10}
        >
          Log Out
        </Button>
      </VStack>
    </StatusContainer>
  );
};

export default SettingsPage;
