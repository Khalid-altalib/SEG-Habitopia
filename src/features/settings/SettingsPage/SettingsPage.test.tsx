import React from "react";
import { fireEvent, render } from "@testing-library/react-native";
import SettingsPage from "./SettingsPage";
import TestingWrapperNavigation from "@app/testingWrapperWithNavigation";
import settingsMockState from "../settingsMockState";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { act } from "react-test-renderer";
import { logOutUser } from "@features/auth/authSlice";

jest.mock("../../auth/authSlice", () => ({
  logOutUser: jest.fn(),
}));

describe("SettingsPage", () => {
  const mockState = {
    settings: settingsMockState,
  };

  const store = configureStore([thunk])(mockState);

  let wrapper: any;

  beforeEach(() => {
    wrapper = render(
      <TestingWrapperNavigation store={store}>
        <SettingsPage />
      </TestingWrapperNavigation>
    );
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls logOutUser when "Log out" button is pressed', () => {
    const button = wrapper.getByText("Log out");

    act(() => {
      fireEvent.press(button);
    });

    expect(logOutUser).toHaveBeenCalledTimes(1);
  });
});
