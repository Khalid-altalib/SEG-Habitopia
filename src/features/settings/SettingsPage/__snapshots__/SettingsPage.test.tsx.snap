// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsPage renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        forwardedRef={[Function]}
        gestureResponseDistance={
          {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        nativeBackButtonDismissalEnabled={false}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onHeaderBackButtonClicked={[Function]}
        onNativeDismissCancelled={[Function]}
        onTransitionProgress={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        stackPresentation="push"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        swipeDirection="horizontal"
      >
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backgroundColor="rgb(255, 255, 255)"
          color="rgb(0, 122, 255)"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={false}
          hideBackButton={false}
          largeTitleHideShadow={false}
          title="TestScreen"
          titleColor="rgb(28, 28, 30)"
          topInsetEnabled={false}
          translucent={false}
        />
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            {
              "flex": 1,
              "flexDirection": "column-reverse",
            }
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                undefined,
              ]
            }
          >
            <View
              dataSet={{}}
              style={{}}
            >
              <View
                dataSet={{}}
                style={{}}
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "paddingBottom": 12,
                    }
                  }
                >
                  <View
                    dataSet={{}}
                    style={{}}
                  >
                    <View
                      dataSet={{}}
                      style={
                        {
                          "alignItems": "flex-start",
                          "display": "flex",
                          "flexDirection": "row",
                        }
                      }
                    >
                      <View
                        dataSet={{}}
                        style={
                          {
                            "borderBottomWidth": 1,
                            "borderColor": "#d4d4d8",
                            "flexDirection": "row",
                          }
                        }
                      >
                        <Text
                          style={
                            [
                              {
                                "color": "#18181b",
                              },
                              {
                                "fontFamily": undefined,
                                "fontSize": 15,
                              },
                              {
                                "flex": 1,
                              },
                            ]
                          }
                        >
                          Name
                        </Text>
                        <View
                          dataSet={{}}
                          style={
                            {
                              "flex": 3,
                              "paddingBottom": 12,
                            }
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              {
                                "opacity": 1,
                              }
                            }
                          >
                            <Text
                              style={
                                [
                                  {
                                    "color": "#18181b",
                                  },
                                  {
                                    "color": "#71717a",
                                    "fontFamily": undefined,
                                    "fontSize": 15,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              Test User
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  dataSet={{}}
                  style={{}}
                />
                <View
                  dataSet={{}}
                  style={
                    {
                      "paddingBottom": 12,
                    }
                  }
                >
                  <View
                    dataSet={{}}
                    style={{}}
                  >
                    <View
                      dataSet={{}}
                      style={
                        {
                          "alignItems": "flex-start",
                          "display": "flex",
                          "flexDirection": "row",
                        }
                      }
                    >
                      <View
                        dataSet={{}}
                        style={
                          {
                            "borderBottomWidth": 1,
                            "borderColor": "#d4d4d8",
                            "flexDirection": "row",
                          }
                        }
                      >
                        <Text
                          style={
                            [
                              {
                                "color": "#18181b",
                              },
                              {
                                "fontFamily": undefined,
                                "fontSize": 15,
                              },
                              {
                                "flex": 1,
                              },
                            ]
                          }
                        >
                          Password
                        </Text>
                        <View
                          dataSet={{}}
                          style={
                            {
                              "flex": 3,
                              "paddingBottom": 12,
                            }
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              {
                                "opacity": 1,
                              }
                            }
                          >
                            <Text
                              style={
                                [
                                  {
                                    "color": "#18181b",
                                  },
                                  {
                                    "color": "#71717a",
                                    "fontFamily": undefined,
                                    "fontSize": 15,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              *******
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  dataSet={{}}
                  style={{}}
                />
                <View
                  dataSet={{}}
                  style={
                    {
                      "paddingBottom": 12,
                    }
                  }
                >
                  <View
                    dataSet={{}}
                    style={{}}
                  >
                    <View
                      dataSet={{}}
                      style={
                        {
                          "alignItems": "flex-start",
                          "display": "flex",
                          "flexDirection": "row",
                        }
                      }
                    >
                      <View
                        dataSet={{}}
                        style={
                          {
                            "borderBottomWidth": 1,
                            "borderColor": "#d4d4d8",
                            "flexDirection": "row",
                          }
                        }
                      >
                        <Text
                          style={
                            [
                              {
                                "color": "#18181b",
                              },
                              {
                                "fontFamily": undefined,
                                "fontSize": 15,
                              },
                              {
                                "flex": 1,
                              },
                            ]
                          }
                        >
                          Bio
                        </Text>
                        <View
                          dataSet={{}}
                          style={
                            {
                              "flex": 3,
                              "paddingBottom": 12,
                            }
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              {
                                "opacity": 1,
                              }
                            }
                          >
                            <Text
                              style={
                                [
                                  {
                                    "color": "#18181b",
                                  },
                                  {
                                    "color": "#71717a",
                                    "fontFamily": undefined,
                                    "fontSize": 15,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              Hello, I'm Test User.
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  dataSet={{}}
                  style={{}}
                />
                <View
                  dataSet={{}}
                  style={
                    {
                      "alignItems": "center",
                      "borderBottomWidth": 1,
                      "borderColor": "#d4d4d8",
                      "display": "flex",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "paddingBottom": 12,
                    }
                  }
                >
                  <View
                    dataSet={{}}
                    style={{}}
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "#18181b",
                          },
                          {
                            "fontFamily": undefined,
                            "fontSize": 15,
                          },
                          undefined,
                        ]
                      }
                    >
                      Notifications 
                    </Text>
                    <Text
                      style={
                        [
                          {
                            "color": "#18181b",
                          },
                          {
                            "color": "#71717a",
                            "fontFamily": undefined,
                            "fontSize": 15,
                          },
                          undefined,
                        ]
                      }
                    >
                      Choose to receive notifications
                    </Text>
                  </View>
                  <View
                    dataSet={{}}
                    style={{}}
                  />
                  <View
                    dataSet={{}}
                    style={{}}
                  >
                    <RCTSwitch
                      accessibilityRole="switch"
                      onChange={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      onTintColor="#7e22ce"
                      style={
                        {
                          "height": 31,
                          "width": 51,
                        }
                      }
                      testID="switch"
                      thumbTintColor="white"
                      value={true}
                    />
                  </View>
                </View>
                <View
                  dataSet={{}}
                  style={{}}
                />
                <View
                  dataSet={{}}
                  style={
                    {
                      "alignItems": "center",
                      "display": "flex",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "paddingTop": 12,
                    }
                  }
                >
                  <View
                    dataSet={{}}
                    style={{}}
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "#18181b",
                          },
                          {
                            "fontFamily": undefined,
                            "fontSize": 15,
                          },
                          undefined,
                        ]
                      }
                    >
                      Light Mode 
                    </Text>
                    <Text
                      style={
                        [
                          {
                            "color": "#18181b",
                          },
                          {
                            "color": "#71717a",
                            "fontFamily": undefined,
                            "fontSize": 15,
                          },
                          undefined,
                        ]
                      }
                    >
                      Choose the color mode
                    </Text>
                  </View>
                  <View
                    dataSet={{}}
                    style={{}}
                  />
                  <View
                    dataSet={{}}
                    style={{}}
                  >
                    <RCTSwitch
                      accessibilityRole="switch"
                      onChange={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      onTintColor="#7e22ce"
                      style={
                        {
                          "height": 31,
                          "width": 51,
                        }
                      }
                      thumbTintColor="white"
                      value={true}
                    />
                  </View>
                </View>
                <View
                  dataSet={{}}
                  style={{}}
                />
                <View
                  isFullWidth={true}
                  onPress={[Function]}
                  style={
                    {
                      "marginTop": 40,
                    }
                  }
                  type={0}
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "marginTop": 40,
                        "opacity": 1,
                      }
                    }
                  >
                    <ViewManagerAdapter_ExpoLinearGradient
                      proxiedProperties={
                        {
                          "colors": [
                            4278220774,
                            4286456526,
                          ],
                          "endPoint": [
                            1,
                            1,
                          ],
                          "locations": undefined,
                          "startPoint": [
                            0,
                            0,
                          ],
                        }
                      }
                      style={
                        [
                          {
                            "alignItems": "center",
                            "borderRadius": 10,
                            "flexDirection": "row",
                            "justifyContent": "center",
                            "paddingHorizontal": 40,
                            "paddingVertical": 15,
                          },
                          {
                            "width": "100%",
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#18181b",
                            },
                            {
                              "color": "#e4e4e7",
                              "fontFamily": undefined,
                              "fontSize": 19,
                            },
                            [
                              null,
                              {
                                "alignSelf": "center",
                              },
                            ],
                          ]
                        }
                      >
                        Log out
                      </Text>
                      <View
                        style={
                          {
                            "marginLeft": 5,
                          }
                        }
                      />
                    </ViewManagerAdapter_ExpoLinearGradient>
                  </View>
                </View>
              </View>
              <View
                dataSet={{}}
                style={
                  {
                    "alignItems": "center",
                    "display": "flex",
                    "justifyContent": "center",
                  }
                }
              />
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenStack>
  </View>
</RNCSafeAreaProvider>
`;
